src/ir/names.h

```diff
@@ -88,6 +87,23 @@ inline Name getValidElementSegmentName(Module& module, Name root) {
     root, [&](Name test) { return !module.getElementSegmentOrNull(test); });
 }
 
+class MinifiedNameGenerator {
+  // Reserved words we must not emit.
+  static std::unordered_set<std::string> reserved;
+
+  // Possible initial letters.
+  static std::string validInitialChars;
+
+  // Possible later letters.
+  static std::string validLaterChars;
```

Comment:
Do the statics need to be in the header? They could be only in the cpp I think, which seems better to me.